app/main.py

```diff
@@ -1,8 +1,31 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self, comfort_class: int, clean_mark: int, brand: str) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self, distance_from_city_center: float, clean_power: int, average_rating: float,
+                 count_of_ratings: int) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list[Car]):
+        income = 0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                income += car.comfort_class * (self.clean_power - car.clean_mark) * self.average_rating / self.distance_from_city_center
+                car.clean_mark = self.clean_power
+        return round(income, 1)
+
+    def calculate_washing_price(self, car: Car):
+        if car.clean_mark < self.clean_power:
+            return round(car.comfort_class * (self.clean_power - car.clean_mark) * self.average_rating / self.distance_from_city_center, 1)
+        return 0
+
+    def rate_service(self, num: int):
+        self.average_rating = round(((self.count_of_ratings * self.average_rating + num) / (self.count_of_ratings + 1)), 1)
```

Comment:
In the `rate_service` method, ensure that the `num` parameter is within the valid range (e.g., 1 to 5) before updating the average rating. This will prevent invalid ratings from affecting the average.